tests/components/sensor/test_awair.py

```diff
@@ -0,0 +1,281 @@
+"""Tests for the Awair sensor platform."""
+
+from contextlib import contextmanager
+from datetime import timedelta
+import json
+import logging
+from unittest.mock import patch
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.sensor.awair import (
+    ATTR_LAST_API_UPDATE, ATTR_TIMESTAMP, DEVICE_CLASS_CARBON_DIOXIDE,
+    DEVICE_CLASS_PM2_5, DEVICE_CLASS_SCORE,
+    DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS)
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY, DEVICE_CLASS_TEMPERATURE, STATE_UNAVAILABLE,
+    TEMP_CELSIUS)
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import parse_datetime, utcnow
+
+from tests.common import async_fire_time_changed, load_fixture, mock_coro
+
+DISCOVERY_CONFIG = {
+    'sensor': {
+        'platform': 'awair',
+        'access_token': 'qwerty',
+    }
+}
+
+MANUAL_CONFIG = {
+    'sensor': {
+        'platform': 'awair',
+        'access_token': 'qwerty',
+        'devices': [
+            {'uuid': 'awair_foo'}
+        ]
+    }
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+NOW = utcnow()
+AIR_DATA_FIXTURE = json.loads(load_fixture('awair_air_data_latest.json'))
+AIR_DATA_FIXTURE[0][ATTR_TIMESTAMP] = str(NOW)
+AIR_DATA_FIXTURE_UPDATED = json.loads(
+    load_fixture('awair_air_data_latest_updated.json'))
+AIR_DATA_FIXTURE_UPDATED[0][ATTR_TIMESTAMP] = str(NOW + timedelta(minutes=5))
+
+
+@contextmanager
+def alter_time(retval):
+    """Manage multiple time mocks."""
+    patch_one = patch('homeassistant.util.dt.utcnow', return_value=retval)
+    patch_two = patch('homeassistant.util.utcnow', return_value=retval)
+
```

Comment:
Add a third time patch here for `'homeassistant.components.sensor.awair.dt.utcnow'`.

Reply:
:+1: